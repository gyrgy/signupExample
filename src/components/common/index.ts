export { default as ButtonPrimary } from 'src/components/common/buttonPrimary';
export { default as Checkbox } from 'src/components/common/checkbox';
export { default as Content } from 'src/components/common/content';
export { default as Fade } from 'src/components/common/fade';
export { default as Form } from 'src/components/common/form';
export { default as InputField } from 'src/components/common/inputField';
export { default as TabPanel } from 'src/components/common/tabPanel';