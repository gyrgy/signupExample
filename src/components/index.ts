export { default as SignUpDone } from 'src/components/signUpDone';
export { default as SignUpPrivacyForm } from 'src/components/signUpPrivacyForm';
export { default as SignUpTabMenu } from 'src/components/signUpTabMenu';
export { default as SignUpUserForm } from 'src/components/signUpUserForm';
