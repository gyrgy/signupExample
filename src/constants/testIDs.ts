export const primaryButtonTestID = 'components/constants/buttonPrimary';
export const checkboxTestID = 'components/constants/checkbox';
export const contentTestID = 'components/constants/content';
export const fadeTestID = 'components/constants/fade';
export const formTestID = 'components/constants/form';
export const inputFieldTestID = 'components/constants/form';
export const tabPanelTestID = 'components/constants/tabPanel';
export const privacyFormTestID = 'components/signUpPrivacyForm';
export const privacyFormTrayCheckboxTestID =
  'components/signUpPrivacyForm/trayCheckbox';
export const privacyFormOtherCheckboxTestID =
  'components/signUpPrivacyForm/otherCheckbox';
export const tabMenuTestID = 'components/signUpTabMenu';
export const tabMenuUserTabTestID = 'components/signUpTabMenu/userTab';
export const tabMenuPrivacyTabTestID = 'components/signUpTabMenu/privacyTab';
export const tabMenuDoneTabTestID = 'components/signUpTabMenu/doneTab';
export const userFormTestID = 'components/signUpUserForm';
export const userFormNameFieldTestID = 'components/signUpUserForm/nameField';
export const userFormRoleFieldTestID = 'components/signUpUserForm/roleField';
export const userFormEmailFieldTestID = 'components/signUpUserForm/emailField';
export const userFormPasswordFieldTestID =
  'components/signUpUserForm/passwordField';
export const signUpDoneTestID = 'components/signUpDone';
