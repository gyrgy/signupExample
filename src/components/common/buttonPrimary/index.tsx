import React from 'react';
import { Button } from '@material-ui/core';
import { primaryButtonTestID } from 'src/constants/testIDs';

interface PrimaryButtonProps {
  label: string;
  onClick: () => void;
  testID?: string;
}

const ButtonPrimary = (props: PrimaryButtonProps) => {
  const { label, onClick, testID } = props;

  return (
    <Button variant="contained" color="primary" onClick={onClick} data-testid={testID || primaryButtonTestID}>
      {label}
    </Button>
  );
};

export default ButtonPrimary;
