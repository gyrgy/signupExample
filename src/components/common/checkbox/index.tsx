import React, { ChangeEvent } from 'react';
import { FormControlLabel, Checkbox } from '@material-ui/core';
import { checkboxTestID } from 'src/constants/testIDs';

interface InputCheckboxProps {
  name: string;
  label: string;
  checked: boolean;
  onChange: (event: ChangeEvent<HTMLInputElement>) => void;
  testID?: string
}


const InputCheckbox = (props: InputCheckboxProps) => {

  const { name, label, checked, onChange, testID } = props;

  return (
    <FormControlLabel
      control={<Checkbox checked={checked} onChange={onChange} name={name} color="primary" data-testid={testID || checkboxTestID} />}
      label={label}
    />
  );
};

export default InputCheckbox;
