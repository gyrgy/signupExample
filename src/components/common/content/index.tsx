import React, { ReactNode } from 'react';
import { makeStyles, Theme } from '@material-ui/core/styles';
import { contentTestID } from 'src/constants/testIDs';

const useStyles = makeStyles((theme: Theme) => ({
  root: {
    backgroundColor: theme.palette.background.paper,
  },
}));

interface ContentProps {
  children: ReactNode;
  testID?: string;
}

const Content = (props: ContentProps) => {
  const { root } = useStyles();
  const { children, testID } = props;

  return (
    <div className={root} data-testid={testID || contentTestID}>
      {children}
    </div>
  );
};

export default Content;