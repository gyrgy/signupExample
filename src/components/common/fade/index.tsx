import React, { ReactNode } from 'react';
import { Fade } from '@material-ui/core';
import { fadeTestID } from 'src/constants/testIDs';

interface FadeEffectProps {
  fadeIn: boolean;
  timeout?: number;
  children: ReactNode;
  testID?: string;
}

const FadeEffect = (props: FadeEffectProps) => {
  const { fadeIn, timeout, children, testID } = props;

  return (
    <Fade
      in={fadeIn}
      timeout={timeout || 300}
      data-testid={testID || fadeTestID}
    >
      <div>
        {children}
      </div>
    </Fade>
  );
};

export default FadeEffect;