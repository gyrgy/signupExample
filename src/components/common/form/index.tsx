import React, { ReactNode } from 'react';
import { createStyles, makeStyles, Theme } from '@material-ui/core/styles';
import { formTestID } from 'src/constants/testIDs';

interface FormProps {
  children: ReactNode;
  testID?: string;
}

const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    root: {
      '& > *': {
        margin: theme.spacing(1),
      },
    },
  })
);

const Form = (props: FormProps) => {
  const { root } = useStyles();
  const { children, testID } = props;

  return (
    <form className={root} noValidate autoComplete="off" data-testid={testID || formTestID}>
      {children}
    </form>
  );
};

export default Form;
