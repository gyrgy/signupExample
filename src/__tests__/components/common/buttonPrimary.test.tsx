import React from 'react';
import { render, fireEvent, RenderResult } from '@testing-library/react';
import { ButtonPrimary } from 'src/components/common';
import { primaryButtonTestID } from 'src/constants/testIDs';

describe('test primary button', () =>{
  const spyOnClickHandler = jest.fn();
  const testLabel = 'Test Label';
  let buttonComponent: RenderResult;

  beforeEach(() => {
    buttonComponent = render(
      <ButtonPrimary label={testLabel} onClick={spyOnClickHandler} />
    );
  });

  it('tests primary button label', () => {
    const { getByText } = buttonComponent;
    const label = getByText(testLabel);
    expect(label).toBeInTheDocument();
  });

  it('tests button callback', () => {
    const { getByTestId } = buttonComponent;
    const button = getByTestId(primaryButtonTestID);
    fireEvent.click(button);
    expect(spyOnClickHandler).toBeCalledTimes(1);
  });

  it('tests if matches with snapshot', () => {
    const { asFragment } = buttonComponent;
    expect(asFragment()).toMatchSnapshot();
  });
});