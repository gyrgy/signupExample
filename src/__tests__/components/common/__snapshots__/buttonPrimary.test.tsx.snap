// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test primary button tests if matches with snapshot 1`] = `
<DocumentFragment>
  <button
    class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
    data-testid="components/constants/buttonPrimary"
    tabindex="0"
    type="button"
  >
    <span
      class="MuiButton-label"
    >
      Test Label
    </span>
    <span
      class="MuiTouchRipple-root"
    />
  </button>
</DocumentFragment>
`;
